pkg/download/download.go

```diff
@@ -130,6 +143,34 @@ func Download(ctx context.Context, endpoint string, opts ...Option) (b []byte, e
 func download(req *http.Request, dlOpts *downloadOptions) (data []byte, err error) {
 	client := &http.Client{}
 
+	if dlOpts.LowSrcPort {
+		port := 100 + rand.Intn(512)
+
+		localTCPAddr, tcperr := net.ResolveTCPAddr("tcp", ":"+strconv.Itoa(port))
+		if tcperr != nil {
+			return nil, retry.ExpectedError(fmt.Errorf("resolving source tcp address: %s", tcperr.Error()))
+		}
+
+		client.Transport = &http.Transport{
+			Proxy: func(req *http.Request) (*url.URL, error) {
+				return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
+			},
+			DialContext: (&net.Dialer{
+				LocalAddr: localTCPAddr,
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+			ForceAttemptHTTP2:     true,
+			DisableKeepAlives:     true,
+			MaxIdleConnsPerHost:   -1,
+		}
```

Comment:
I've found similar here:

internal/pkg/containers/image/resolver.go:162
internal/app/machined/main.go:46

Reply:
yep, exactly, but you can see it sets a lot of weird defaults

Reply:
so the idea is to get `cleanhttp.DefaultTransport()` and modify it accordingly. It will require some careful typecasting though (unfortunately)